components/sock_utils/test/host/main/test_sock_utils.cpp

```diff
@@ -0,0 +1,139 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include "ifaddrs.h"
+#include "esp_netif.h"
+#include "esp_event.h"
+#include "catch2/catch_test_macros.hpp"
+#include "catch2/catch_session.hpp"
+
+static char buffer[64];
+
+static esp_netif_t *create_test_netif(const char *if_key)
+{
+    esp_netif_inherent_config_t base_cfg = ESP_NETIF_INHERENT_DEFAULT_WIFI_STA();
+    esp_netif_ip_info_t ip = { };
+    ip.ip.addr = ESP_IP4TOADDR(1, 2, 3, 4);
```

Comment:
If IP address is parameter evaluated that address is gotten correctly, I think we should have different address for each interface to distinguish among them.